src/main/java/io/vlingo/symbio/store/object/jdbc/JDBCObjectStoreActor.java

```diff
@@ -24,16 +41,43 @@
  * any number of {@code JDBCObjectStoreDelegate} types.
  */
 public class JDBCObjectStoreActor extends Actor implements ObjectStore, Scheduled<Object> {
+  private final DispatcherControl dispatcherControl;
   private boolean closed;
   private final JDBCObjectStoreDelegate delegate;
+  private final Dispatcher<Dispatchable<BaseEntry.TextEntry, State.TextState>> dispatcher;
+  private final Logger logger;
+  private final EntryAdapterProvider entryAdapterProvider;
+  private final StateAdapterProvider stateAdapterProvider;
+  private final IdentityGenerator identityGenerator;
 
   @SuppressWarnings("unchecked")
-  public JDBCObjectStoreActor(final JDBCObjectStoreDelegate delegate) {
+  public JDBCObjectStoreActor(final JDBCObjectStoreDelegate delegate, final Dispatcher<Dispatchable<BaseEntry.TextEntry, State.TextState>> dispatcher) {
+     this(delegate, dispatcher, 1000L, 1000L);
+  }
+
+  @SuppressWarnings("unchecked")
+  public JDBCObjectStoreActor(final JDBCObjectStoreDelegate delegate, final Dispatcher<Dispatchable<BaseEntry.TextEntry, State.TextState>> dispatcher,
+          final long checkConfirmationExpirationInterval, final long confirmationExpiration) {
     this.delegate = delegate;
+    this.dispatcher = dispatcher;
     this.closed = false;
+    this.logger = stage().world().defaultLogger();
+    this.entryAdapterProvider = EntryAdapterProvider.instance(stage().world());
+    this.stateAdapterProvider = StateAdapterProvider.instance(stage().world());
+    this.identityGenerator = new IdentityGenerator.TimeBasedIdentityGenerator();
```

Comment:
Here and elsewhere, I don't know if the TimeBasedIdentityGenerator buys anything special. Is the internal timestamp actually being used? If not then a version 4 would be more efficient because this version 2 generator synchronizes threads in some/all cases. The default Java UUID is version 4.

Reply:
Changed to `IdentityGenerator.RandomIdentityGenerator()`